### PR TITLE
Added ESP component and SESP (stub ESP model).

### PR DESCRIPTION
**Pull request by gold2718**
- Added an eighth component type, ESP to CIME.
- Added the stub ESP model, SESP.
- Made seq_flds_x2c_fluxes and seq_flds_c2x_fluxes inputs to component_run optional (ESP components do not communicate with coupler).
- Created new interfaces to seq_infodata_getData and seq_infodata_putData which take a component one-letter-id and make the correct retrievals or settings. Use new interfaces in component_mod.
- Use new parameters to clarify how many communicators are needed for a run.
- Use ID = -1 to indicate an idea which is not attached to a seq_comms object.

Test suite:  CIME standalone tests, FC5
Test baseline: cesm1_5_alpha06
Test namelist changes: NA
Test status: bit for bit

Fixes: NA

User interface changes?: Optional ESP component type in compset longname

Code review:  Pending

### CODE REVIEW

File: cime_config/cesm/machines/buildlib.csm_share [hunk 1]
```diff
@@ -26,6 +26,7 @@ my $NINST_LND		= `./xmlquery NINST_LND		-value`;
 my $NINST_OCN		= `./xmlquery NINST_OCN		-value`;
 my $NINST_ROF		= `./xmlquery NINST_ROF		-value`;
 my $NINST_WAV		= `./xmlquery NINST_WAV		-value`;
+my $NINST_ESP		= `./xmlquery NINST_ESP		-value`;
```
Comment (jedwards4b) [inline cime_config/cesm/machines/buildlib.csm_share:29]: Are we allowing for more than one instance of this component?

File: doc/modelnl/env_mach_pes.html [hunk 2]
```diff
@@ -649,6 +649,82 @@
     </td>
     <td>env_mach_pes.xml</td>
   </tr>
+  <tr id="NTASKS_ESP_tr">
```
Comment (jedwards4b) [inline doc/modelnl/env_mach_pes.html:652]: whats all the java script stuff about?

Comment (gold2718) [inline doc/modelnl/env_mach_pes.html:652]: I don't see anything different here. This is just how the file is set up to make it searchable and expandable. The page seems to work fine as is.

File: cime_config/cesm/machines/buildlib.csm_share (hunk 1 above)
Comment (gold2718) [inline cime_config/cesm/machines/buildlib.csm_share:29]: I do not see any reason to restrict it. While the immediate goal is DART (an ensemble filter), future applications, such as geoengineering, may have the same number of instances as the atmosphere.

**Comment by gold2718** [comment]
I added the comments as requested in the code review. @jedwards4b, can you merge now?

**Comment by jedwards4b** [comment]
done

On Mon, Apr 18, 2016 at 5:59 PM, goldy notifications@github.com wrote:

> Assigned #418 https://github.com/CESM-Development/cime/pull/418 to
> @jedwards4b https://github.com/jedwards4b.
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> https://github.com/CESM-Development/cime/pull/418#event-632146480

## 

Jim Edwards

CESM Software Engineer
National Center for Atmospheric Research
Boulder, CO

**Pull request merged**